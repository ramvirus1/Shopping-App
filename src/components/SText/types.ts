export interface ISText {
    text: string | number;
    weight?: 'normal' | 'bold' | 'semibold';
    size?: number;
};

export interface IText {
    weight: string;
    size: number;
}