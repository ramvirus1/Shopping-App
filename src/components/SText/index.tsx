import styled from "styled-components/native";
import { ISText, IText } from "./types";
import { scale } from "react-native-size-matters";

const weightMapping: {[key: string]: string} = {
    bold: " Inter_700Bold",
    semibold: "Inter_600SemiBold",
    normal: "Inter_400Regular"
};

const SText = ({ 
    text,
    weight = "normal",
    size = 10
}: ISText): JSX.Element => {
    return <Text numberOfLines={2} weight={weightMapping[weight]} size={size}>{text}</Text>
};

export default SText;

const Text = styled.Text<IText>`
    font-family: ${p => p.weight};
    font-size: ${p => scale(p.size)}px;
`