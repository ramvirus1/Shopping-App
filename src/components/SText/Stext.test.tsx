import renderer, { ReactTestRendererJSON } from 'react-test-renderer';
import SText from "./index";

const props = {
    text: "Hi",
    size: 20
};

describe('SText', () => {
  const snapshot = renderer.create(<SText {...props} />).toJSON();
  it('should render', () => {
    expect(snapshot).toMatchSnapshot();
  });
});