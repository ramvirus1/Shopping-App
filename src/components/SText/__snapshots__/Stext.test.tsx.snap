// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SText should render 1`] = `
<Text
  numberOfLines={2}
  size={20}
  style={
    Array [
      Object {
        "fontFamily": "Inter_400Regular",
        "fontSize": 42.857142857142854,
      },
    ]
  }
  weight="Inter_400Regular"
>
  Hi
</Text>
`;
